pkg/apis/ml/v1alpha1/extension_spec_job.go

```diff
@@ -25,31 +25,60 @@ import (
 	sharedApi "github.com/arangodb/kube-arangodb/pkg/apis/shared/v1"
 )
 
-type ArangoMLExtensionSpecInit struct {
+type ArangoMLJobsTemplates struct {
+	// Prediction defines template for the prediction job
+	Prediction map[string]*ArangoMLExtensionTemplateSpec `json:"prediction,omitempty"`
+
+	// Training defines template for the training job
+	Training map[string]*ArangoMLExtensionTemplateSpec `json:"training,omitempty"`
+}
+
+func (j *ArangoMLJobsTemplates) Validate() error {
+	if j == nil {
+		return nil
+	}
+
+	var errs []error
+	for _, template := range j.Prediction {
+		if err := template.Validate(); err != nil {
```

Comment:
No need to check for errors - `shared.WithErrors` will combine only non-nil entries anyway